blog/2020-09-23-git-usage.md

```diff
@@ -0,0 +1,209 @@
+---
+id: git-usage
+title: Git 使用指南：理解 Git 的工作原理
+author: Josh Cena
+authorTitle: Leader of Computerization
+authorURL: https://github.com/Josh-Cena
+authorImageURL: https://avatars2.githubusercontent.com/u/55398995?s=400&u=88dc0dcb0691877524dd8739db9fde7ed4fa9721&v=4
+tags: [git, guide]
+---
+
+> 这篇文章是从[C社的新成员练手 repo](https://github.com/Computerization/New-member-practice-commit) 的 README.md 的第一节转移过来的，添加了一些内容，并做了相应的翻译。（原文用英文的原因，应该是sy大佬和我都更习惯用英语写技术相关的内容吧……）
+
+添加文件时，你既可以用带图形用户界面（GUI）的 GitHub Desktop，也可以用命令行。你可以从 GUI 入手，但你会有一天意识到命令行功能的强大，开始用它的。另外，Visual Studio Code 的用户们也可以试试其自带的源代码管理工具。
+
+我们觉得有必要给你解释你每一步究竟在做什么，而不是让你机械地重复我们写好的教程。这对你尤其有帮助，因为教程通常都把事情想得很完美，但现实则充满了各种意外和变数。阅读时，我们并不要求你事先懂得任何 Git 操作。
+
+你可以把 Git 理解成一个版本控制系统。它能记录每个文件的创建、更改、和删除，而仓库管理者可以在各个版本（commit[^commit]）间自由切换，就好像游戏中的若干存档一样。
+
+[^commit]: 中文大致意思是“提交”——这也是我不喜欢用中文写作的原因，中英混杂总是难以避免。
+
+## 基础操作：clone, branch, commit, push, pull request
+
+假设现在 GitHub 上有这样一个叫 `Hello` 的仓库，是由小丽创建的，装着可能是历史上*最*著名的一段代码。它的目录长这个样子：
+
+```bash
+.
+├── Hello.cpp
+│
+├── Contributors.txt
+│
+└── README.md
+```
+
+```C++
+// Hello.cpp
+#include <iostream>
+using namespace std;
+int main() {
+  cout << "Hello world!" << endl;
+}
+```
+
+```bash
+// Contributors.txt
+XiaoLi
+```
+
+```Markdown
+// README.md
+# Hello world
+A piece of C++ code that prints `Hello world!`
+```
+
+你可以把每个 commit 看作是整个仓库的一个快照，一个拷贝。实际上它要比一份完整的拷贝轻量得多，但原理上它保存了那一时刻仓库的全部信息。比如上面的版本就可以看作 `C0` commit。（每个 commit 都有一个独有的哈希值，但长到人类无法阅读。因此用 `C0` 来指代就好了。）我们也管这个 commit 叫 `master` 分支。*一个分支即一系列有线性发展关系的 commit，而 `master` 分支则作为主分支。实际上，一个分支就是一个指向某个 commit 的指针。* 该项目目前的树状结构如下：
+
+<img alt="Git tree 1" src={useBaseUrl('img/blog/2020-09-23/Git-tree-1.png')} />
```

Comment:
You'll have to first import `useBaseUrl` 
```
import useBaseUrl from '@docusaurus/useBaseUrl';
```

See https://v2.docusaurus.io/docs/static-assets/#markdown-example